docs/src/creating-packages.md

```diff
@@ -299,6 +299,11 @@ loaded into the Julia session. This is very similar to functionality that the ex
 [Requires.jl](https://github.com/JuliaPackaging/Requires.jl) provides, but which is now available directly through Julia,
 and provides added benefits such as being able to precompile the extension.
 
+!!! warning
+    The recommended way to use extensions is to implement new methods for _existing functions_.
+    This can include zero-method functions defined in the main package by `function f end`.
+    On the other hand, if you create _new objects_ like functions or types in the extension, it is possible to access them (see below) but it is not considered good practice.
+
```

Comment:
I'd perhaps have put this paragraph a little later in the documentation, perhaps even as far down as just before the second newly added paragraph about new functions/types.

But in any case, I feel this warning is more easily understood once the reader has seen an example of extensions in action. This could also allow referring to the code example inside the warning text. For example:

> The recommended way to use extensions is to implement new methods for _existing functions_ (as is the case of `Plotting.plot` in the example above). This can include...